### PR TITLE
fix(github): Catch errors inside `fetchPr` call

### PR DESCRIPTION
**Pull request by zharinov**
## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/platform/github/index.ts [hunk 1]
```diff
@@ -582,12 +582,17 @@ function cachePr(pr?: Pr | null): void {
...
-  return result;
+  try {
+    const { body: ghRestPr } = await githubApi.getJson<GhRestPr>(
+      `repos/${config.parentRepo ?? config.repository}/pulls/${prNo}`
+    );
+    const result = coerceRestPr(ghRestPr);
+    cachePr(result);
+    return result;
+  } catch (err) {
+    logger.debug({ err }, `fetchPr(${prNo}) error`);
```
Comment (rarkins) [inline lib/modules/platform/github/index.ts:593]: Should this be a warn? If so then we want it to have a constant string for the message, and move the prNo to the metadata

**Review by rarkins — DISMISSED** [review]

File: lib/modules/platform/github/common.ts [hunk 2]
```diff
@@ -8,6 +8,7 @@ import type { GhRestPr } from './types';
  * @see https://docs.github.com/en/rest/reference/pulls#list-pull-requests
  */
 export function coerceRestPr(pr: GhRestPr | null | undefined): Pr | null {
+  // istanbul ignore if
```
Comment (viceice) [inline lib/modules/platform/github/common.ts:11]: should have a test.

**Review by viceice — COMMENTED** [review]
otherwise LGTM

File: lib/modules/platform/github/common.ts (hunk 2 above)
Comment (zharinov) [reply lib/modules/platform/github/common.ts:11]: I'm about to remove this optionality later

**Review by rarkins — APPROVED** [review]

**Pull request merged**